stage/writing_stage.py

```diff
@@ -0,0 +1,34 @@
+"""Writing to the fuzz target and build script."""
+import logging
+
+from result_classes import Result
+from stage.base_stage import BaseStage
+
+logging.basicConfig(level=logging.DEBUG,
+                    format=('%(asctime)s [PID: %(process)d] %(levelname)s '
+                            '[%(module)s.%(funcName)s]: %(message)s'))
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+class WritingStage(BaseStage):
+  """Writing to the fuzz target and build script."""
+
+  def _write_new_fuzz_target(self, prev_results: list[Result]) -> Result:
+    """Writes a new fuzz target."""
+    return self.get_agent('Prototyper').execute(prev_results)
```

Comment:
If particular stages expect agents of certain types to exist, should we ensure they are passed in the constructor? that way, we can never get into a state where they don't exist.

Reply:
Done, now we pass all agents expected by each stage in the pipeline constructor.